lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -413,6 +413,45 @@ class UnsqueezeOpConversionPattern
   }
 };
 
+class ConstantOpConversionPattern
+    : public OpConversionPattern<ttir::ConstantOp> {
+public:
+  using OpConversionPattern<ttir::ConstantOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::ConstantOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    ::mlir::ElementsAttr valueAttr = op.getValue();
+    if (!valueAttr.getElementType().isIntOrFloat()) {
+      return rewriter.notifyMatchFailure(
+          op, "TTNN doesn't currently support tensor creation from values "
```

Comment:
Can you please move all checks to checkBasicLegality() to keep the check consistent for all patterns.